spec/features/users/sign_in_spec.rb

```diff
@@ -859,10 +860,21 @@
 
       expect(current_path).to eq(account_path)
       doc_auth_log = DocAuthLog.find_by(user_id: user.id)
-      expect(doc_auth_log.no_sp_campaign).to eq(campaign)
+      expect(doc_auth_log.no_sp_campaign).to eq(ok_campaign)
       expect(doc_auth_log.no_sp_session_started_at).to be_present
     end
 
+    it 'does not save a bad campaign', :bad_campaign do
```

Comment:
I think this would be a simpler test with a `let(:campaign) { bad_campaign }` than an attribute, like:

```ruby
context 'with a bad campaign' do
  let(:campaign) { bad_campaign }
  it 'does not save' do
    # ...
```

Reply:
nice!